bokeh/core/properties.py

```diff
@@ -1255,6 +1263,23 @@ def from_json(self, json, models=None):
         else:
             raise DeserializationError("%s expected a dict or None, got %s" % (self, json))
 
+class ColumnData(Dict):
+    """Property holding column data in form of a dict. Also applies
+    encoding and decoding to the data.
+    """
```

Comment:
Since it's not straightforward to move both the encoding and decoding on ``ColumnArray`` (because property descriptors aren't called during serialization), I've gone with your second suggestion in the end.